packages/angular/src/builders/webpack-server/webpack-server.impl.ts

```diff
@@ -55,22 +55,21 @@ export function webpackServer(schema: Schema, context: BuilderContext) {
               pathToWebpackConfig,
               buildTarget.options.tsConfig
             );
+            // The extra Webpack configuration file can also export a Promise, for instance:
+            // `module.exports = new Promise(...)`. If it exports a single object, but not a Promise,
+            // then await will just resolve that object.
+            const config = await customWebpackConfiguration;
```

Comment:
nit: I preferred the previous version, not sure what prompted this change. In the scenario the `customWebpackConfiguration` is a function, the function declaration doesn't need to be awaited. The previous code seemed more semantically correct. If the intention is to await here the execution of the function, that would need to be done when invoking the function, in this line would be a no-op.

Reply:
The problem with the previous was if the resolution of the promise is a function, then we arent handling it correctly.

i.e.

```ts
config = Promise.resolve((config) => { // do stuff });
```

Config isn't a function, it's a promise.
During execution, we check if its a function, so that we can use the function syntax of webpack correctly.
But this fails, as it's a promise.

then we get to the next step, which will await the promise, but our code is expecting it to now be an object, but it isn't it's a function.

awaiting a non-promise is a no-op anyway, so there's no incorrect behaviour.


Although I might have forgotten to add this logic to webpack-browser

Reply:
Oh, I see, that's fine. I didn't mean it was incorrect, I just missed this use case, so it seemed semantically wrong, but it's all good 👍🏻